src/Node/Node.php

```diff
@@ -235,6 +237,19 @@ public function getTemplateName()
         return $this->name;
     }
 
+    public function setSourceContext(Source $source)
+    {
+        $this->sourceContext = $source;
+        foreach ($this->nodes as $node) {
```

Comment:
In 1.x, you need to account for `null` and for custom implementations of the interface (which would not have the method) here. Both are deprecated, but this code should not break the BC layer.

Reply:
correct